bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,129 @@
+from geopy.geocoders import Nominatim
+from timezonefinder import TimezoneFinder
+import pytz
+from pytz import country_timezones, timezone
+from datetime import datetime
+from discord.ext.commands import Cog, command
+import requests
+from discord import Embed
+import time
+import string
+
+from discord.utils import parse_time
+
+class WeatherCog(Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @command(name='weather', help='Show weather information in your city', usage="{city}")
+    async def my_command(self, ctx, *city_input):
+        # get all argument inputed by the user 
+        # on discord and join them together 
+        # (handle double-words/argument cities)
+        city = " "
+        city = string.capwords(city.join(city_input))
+    
+        # initialize Nominatim API
+        geolocator = Nominatim(user_agent="geoapiExercises")
+        # city_name = city.capitalize()
+        # getting Latitude and Longitude
+        location = geolocator.geocode(city)
+    
+        # pass the Latitude and Longitude
+        # into a timezone_at
+        # and it return timezone
+        obj = TimezoneFinder()
+
+        # returns timezone
+        result = obj.timezone_at(lng=location.longitude, lat=location.latitude)
+        timezone_city = datetime.now(pytz.timezone(result))
+        timezone_city = timezone_city.strftime('%m/%d/%Y %H:%M')
+
+        # Enter your API key here
+        api_key="441df3a5cadc2498e093c0367cae6817"
+        # base_url variable to store url
+        base_url = "http://api.openweathermap.org/data/2.5/weather?"
+        # Give city name
+
+        city_name = str(city)
+        # complete_url variable to store complete url address
+        complete_url = base_url + "appid=" + api_key + "&q=" + city_name
+        # get method of requests module return response object
+        response = requests.get(complete_url)
+        # json method of response object convert json format data into python format data
+        data_weather = response.json()
+
+        # Now data_weather contains list of nested dictionaries, check the value of "cod" key is not equal to
+        # "404", means city is found otherwise, city is not found
+        if data_weather["cod"] != "404":
```

Comment:
Why not reverse the logic here, so `data_weather["cod"] == "200"`. Positive are easier to read.